intel build no longer work with intel-mpi/2019.4 on Grizzly

I've been trying to build with `ifort` from the Makefile and `intel-mpi`.  This used to work on Grizzly but doesn't seem to work anymore. The error when trying to build the PIO2 test program is:
```
ld: cannot find -lmpif90
```

However, if I change `mpif90`, `mpicc` and `mpicxx` to `mpiifort`, `mpiicc` and `mpicpc`, I am able to build successfully but I get errors at runtime:
```
[gr0250:32576] *** An error occurred in MPI_Info_dup
[gr0250:32576] *** reported by process [2785673216,0]
[gr0250:32576] *** on communicator MPI_COMM_WORLD
[gr0250:32576] *** MPI_ERR_INFO: invalid info object
[gr0250:32576] *** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
[gr0250:32576] ***    and potentially your MPI job)
```

Additionally please enter this information if applicable:
 - Compiler and library stack:
```
module purge
module load git
module load friendly-testing
module load intel/19.0.4 intel-mpi/2019.4 hdf5-parallel/1.8.16 pnetcdf/1.11.2 \
    netcdf-h5parallel/4.7.3 mkl/2019.0.4
export NETCDF=/usr/projects/hpcsoft/toss3/grizzly/netcdf/4.7.3_intel-19.0.4_intel-mpi-2019.4_hdf5-1.8.16
export NETCDFF=/usr/projects/hpcsoft/toss3/grizzly/netcdf/4.7.3_intel-19.0.4_intel-mpi-2019.4_hdf5-1.8.16
export PNETCDF=/usr/projects/hpcsoft/toss3/grizzly/pnetcdf/1.11.2_intel-19.0.4_intel-mpi-2019.4_hdf5-1.8.16
export PIO=/usr/projects/climate/xylar/libraries/grizzly/scorpio-1.1.5-intel
export I_MPI_CC=icc
export I_MPI_CXX=icpc
export I_MPI_F77=ifort
export I_MPI_F90=ifort
export CORE=ocean
export USE_PIO2=true
export AUTOCLEAN=true
export  HDF5_USE_FILE_LOCKING=FALSE
```
 - Test case used: `ocean/baroclinic_channel/10km/default`

--- a/Makefile
+++ b/Makefile
@@ -285,6 +285,33 @@ ifort-gcc:
 	"OPENMP = $(OPENMP)" \
 	"CPPFLAGS = $(MODEL_FORMULATION) -D_MPI" )
 
+intel-mpi:
+	( $(MAKE) all \
+	"FC_PARALLEL = mpiifort" \
+	"CC_PARALLEL = mpiicc" \
+	"CXX_PARALLEL = mpiicpc" \
+	"FC_SERIAL = ifort" \
+	"CC_SERIAL = icc" \
+	"CXX_SERIAL = icpc" \
+	"FFLAGS_PROMOTION = -real-size 64" \
+	"FFLAGS_OPT = -O3 -convert big_endian -free -align array64byte" \
+	"CFLAGS_OPT = -O3" \
+	"CXXFLAGS_OPT = -O3" \
+	"LDFLAGS_OPT = -O3" \
+	"FFLAGS_DEBUG = -g -convert big_endian -free -CU -CB -check all -fpe0 -traceback" \
+	"CFLAGS_DEBUG = -g -traceback" \
+	"CXXFLAGS_DEBUG = -g -traceback" \
+	"LDFLAGS_DEBUG = -g -fpe0 -traceback" \
+	"FFLAGS_OMP = -qopenmp" \
+	"CFLAGS_OMP = -qopenmp" \
+	"PICFLAG = -fpic" \
+	"BUILD_TARGET = $(@)" \
+	"CORE = $(CORE)" \
+	"DEBUG = $(DEBUG)" \
+	"USE_PAPI = $(USE_PAPI)" \
+	"OPENMP = $(OPENMP)" \
+	"CPPFLAGS = $(MODEL_FORMULATION) -D_MPI" )
+
 gfortran:
 	( $(MAKE) all \
 	"FC_PARALLEL = mpif90" \

